micrometer-spring-legacy/src/main/java/io/micrometer/spring/MeterRegistryConfigurationSupport.java

```diff
@@ -41,6 +42,8 @@
 
     @PostConstruct
     void bindAll() {
+        MeterFactory.Config.setRegistry(registry);
```

Comment:
Yes and no. It won't break any existing tests unless the test is trying to validate a metric that is being set via the static `MeterFactory` approach. 

Tests that are relying on parallelism/isolation should be using a `MeterProvider` that is injected and guaranteed to match the current application context.